src/haddock/modules/analysis/alascan/scan.py

```diff
@@ -0,0 +1,591 @@
+                f.write(f"# `alascan` cluster results for cluster {cl_id}{os.linesep}")  # noqa E501
+                f.write(f"#{os.linesep}")
+                f.write(f"# z_score is calculated with respect to the mean values of all residues{os.linesep}")  # noqa E501
+                f.write(f"#######################################################################{os.linesep}")  # noqa E501
+                f.write(fl_content)
+    return clt_scan
+
+def check_alascan_jobs(alascan_jobs):
+    """Check that all the alascan jobs have been completed.
+    
+    Parameters
+    ----------
+    alascan_jobs : list
+        List of ScanJob objects.
+    
+    Returns
+    ------
+    alascan_file_l : list
+        List of the alascan files.
+    """
+    alascan_file_l = []
+    not_found = []
+    for job in alascan_jobs:
+        if not job.output.exists():
+            not_found.append(job.output.name)
+            wrn = f'Alascan not completed for {job.output.name}'
+            log.warning(wrn)
+        else:
+            alascan_file_l.append(str(job.output))
+    if not_found:
+        # Not all alascan were executed, cannot proceed
+        raise Exception("Several alascan files were not generated:"
+                               f" {not_found}")
+    
+    return alascan_file_l
+
+def generate_alascan_output(models, path):
+    """Generate the alascan output files."""
+    models_to_export = []
+    for model in models:
+        name = f"{model.file_name.rstrip('.pdb')}_alascan.pdb"
+        # changing attributes
+        name_path = Path(name)
+        shutil.copy(Path(model.path, model.file_name), name_path) 
+        alascan_fname = f"scan_{model.file_name.rstrip('.pdb')}.csv"
+        # add delta_score as a bfactor to the model
+        df_scan = pd.read_csv(alascan_fname, sep="\t", comment="#")
+        add_delta_to_bfactor(name, df_scan)
+        model.ori_name = model.file_name
+        model.file_name = name
+        model.full_name = name
+        model.rel_path = Path('..', Path(path).name, name)
+        model.path = str(Path(path).resolve())
+        models_to_export.append(model)
+    return models_to_export
+
+
+def create_alascan_plots(clt_alascan, scan_residue):
+    """Create the alascan plots."""
+    for clt_id in clt_alascan:
+        scan_clt_filename = f"scan_clt_{clt_id}.csv"
+        if not os.path.exists(scan_clt_filename):
+            log.warning(f"Could not find {scan_clt_filename}")
+            continue
+        df_scan_clt = pd.read_csv(
+            scan_clt_filename,
+            sep="\t",
+            comment="#"
+            )
+        # plot the data
+        try:
+            make_alascan_plot(
+                df_scan_clt,
+                clt_id,
+                scan_residue
+                )
+        except Exception as e:
+            log.warning(
+                "Could not create interactive plot. The following error"
+                f" occurred {e}"
+                )
```

Comment:
This should `raise` an exception and you handle it upstream

Reply:
not sure about this..maybe the code fails in creating a single plot (for some crazy reason) but all the others can be generated correctly. I think it makes sense to handle this scenario at this level